exporter/zipkinexporter/zipkin.go

```diff
@@ -171,8 +204,47 @@ func (ze *zipkinExporter) Start(host exporter.Host) error {
 func (ze *zipkinExporter) Shutdown() error {
 	ze.mu.Lock()
 	defer ze.mu.Unlock()
+	return nil
+}
 
-	return ze.reporter.Close()
+func (ze *zipkinExporter) PushTraceData(ctx context.Context, td consumerdata.TraceData) (droppedSpans int, err error) {
+	tbatch := []*zipkinmodel.SpanModel{}
+	for _, span := range td.Spans {
+		sd, err := spandatatranslator.ProtoSpanToOCSpanData(span)
+		if err != nil {
+			// return len(td.Spans) - goodSpans, consumererror.Permanent(err)
+			return len(td.Spans), consumererror.Permanent(err)
+		}
+		zs := ze.zipkinSpan(td.Node, sd)
+		tbatch = append(tbatch, &zs)
+	}
+
+	body, err := ze.serializer.Serialize(tbatch)
+	if err != nil {
+		// r.logger.Printf("failed when marshalling the spans batch: %s\n", err.Error())
+		return len(td.Spans), err
+	}
+
+	req, err := http.NewRequest("POST", ze.url, bytes.NewReader(body))
+	if err != nil {
+		// r.logger.Printf("failed when creating the request: %s\n", err.Error())
+		return len(td.Spans), err
+	}
+	req.Header.Set("Content-Type", ze.serializer.ContentType())
+	/*
+		if r.reqCallback != nil {
+			r.reqCallback(req)
+		}*/
+
+	resp, err := ze.client.Do(req)
+	if err != nil {
+		return len(td.Spans), err
+	}
+	_ = resp.Body.Close()
+	if resp.StatusCode != 202 {
+		return len(td.Spans), fmt.Errorf("failed the request with status code %d", resp.StatusCode)
+	}
+	return 0, nil
 }
 
 func (ze *zipkinExporter) ConsumeTraceData(ctx context.Context, td consumerdata.TraceData) (zerr error) {
```

Comment:
@pjanotti , this method could be removed. But to make the `zipkin_test` compilable, I did not remove it.

Reply:
You should be able to replace that with a call to `PushTraceData`, you may need a cast to the `zipkinExporter` type but it should be doable.